bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java

```diff
@@ -228,13 +235,14 @@ private void initializeThing() throws IOException {
         profile = tmpPrf; // all initialization done, so keep the profile
 
         // Validate device mode
-        String thingType = getThing().getThingTypeUID().getId();
         String reqMode = thingType.contains("-") ? StringUtils.substringAfter(thingType, "-") : "";
         if (!reqMode.isEmpty() && !tmpPrf.mode.equals(reqMode)) {
+            logger.info(
```

Comment:
Why did you remove the message from the Thing status? The Thing status is the correct place for it.

Reply:
My understanding is that
updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                    "@text/offline.conf-error-wrong-mode")
pulls up the language resource "offline.conf-error-wrong-mode" to get this messages from shelly_de.properties/shelly.properties. Isn't it this the case?

So a info message goes to the log and the thing status gets updated with the genral message - or is there a way to pass parameters to the messages retrived from the properties file?

Reply:
> is there a way to pass parameters to the messages retrived from the properties file?

There is, see https://www.openhab.org/docs/developer/utils/i18n.html#i18n-text-provider-api.

Reply:
ok, I figured out the i18nProvider stuff, but how to format the message in the properties file to inject parameters?

Reply:
@kaikreuzer could you point me to the solution?